libr/bin/p/bin_smd.c

```diff
@@ -149,7 +149,12 @@ static RList *symbols(RBinFile *bf) {
 	}
 	ret->free = free;
 	// TODO: store all this stuff in SDB
-	hdr = (SMD_Header *) (bf->buf->buf + 0x100);
+	SMD_Header shdr = {{0}};
```

Comment:
why a new var? BTW, you don't need to 0 it, because you'll read always at least enough bytes to fill the entire SMD_Header structure, so all fields will be initialized.